clients/src/test/java/org/apache/kafka/clients/consumer/RangeAssignorTest.java

```diff
@@ -0,0 +1,244 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer;
+
+import org.apache.kafka.clients.consumer.internals.AbstractPartitionAssignor;
+import org.apache.kafka.common.TopicPartition;
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class RangeAssignorTest {
+
+    private ByteBuffer topicHash = ByteBuffer.wrap(new byte[] {});
+
+    @Test
+    public void testOneConsumerNoTopic() {
+        String consumerId = "consumer";
+        RangeAssignor assignor = new RangeAssignor();
+        AbstractPartitionAssignor.ConsumerMetadata metadata = new AbstractPartitionAssignor.ConsumerMetadata(
+                Collections.<String>emptyList(), topicHash);
+        Map<String, Integer> partitionsPerTopic = new HashMap<>();
+
+        List<TopicPartition> assignment = assignor.assign(consumerId, partitionsPerTopic,
+                Collections.singletonMap(consumerId, metadata));
+        assertTrue(assignment.isEmpty());
+    }
+
+    @Test
+    public void testOneConsumerNonexistentTopic() {
+        String topic = "topic";
+        String consumerId = "consumer";
+        RangeAssignor assignor = new RangeAssignor();
+        AbstractPartitionAssignor.ConsumerMetadata metadata = new AbstractPartitionAssignor.ConsumerMetadata(
+                Arrays.asList(topic), topicHash);
+        Map<String, Integer> partitionsPerTopic = new HashMap<>();
```

Comment:
Not a big deal, but `Collections.singletonMap` is a nice shortcut for building maps like this.